R/otel.R

```diff
@@ -130,7 +127,7 @@ with_ospan_async <- function(
   }
   on.exit(if (needs_cleanup) cleanup(), add = TRUE)
 
-  result <- with_ospan_promise_domain(span, expr)
+  result <- with_active_span(span, expr)
```

Comment:
```suggestion
  result <- with_ospan_promise_domain(with_active_span(span, expr))
```

This would NOT be needed in Shiny (since there's already a `with_ospan_promise_domain` wrapping everything) but it WOULD be needed in any non-Shiny code that uses otel and promises, right?